### PR TITLE
Adding `messageOptions` to requests

### PR DESCRIPTION
**Pull request by poketch**
As discussed: https://discord.com/channels/170995199584108546/722566083499786352/1146928714827956394

I couldn't think of an applicable demo but if oyu have an idea for one I can add that too

### CODE REVIEW

**Comment by krbz999** [comment]
The changes look good to me - but does this actually work? My understanding was `blind` did nothing unless the ChatMessage contained a Roll.

**Comment by poketch** [comment]
Ah, you're correct. I didn't know that, was dead tired last night and threw this together hastily without properly testing (in a clean environment), looks like MidiQOL is doing 
```js
Hooks.on("renderChatMessage", (msg, html, data) => {
      if (enablingSetting && msg.blind && !game.user?.isGM) {
    html.hide();
    return;
  }
});
```
under the hood which is what was hiding the blind requests for me. 
This probably puts this out of scope for what you want to do with Requestor.

That being said, if you're interested, I think I can generalize this change by introducing a `messageOptions` object which is merged onto your constructed `messageData` object before passing to `ChatMessage.create()`, this would allow users to pass whatever options they please allowing them to make changes suited for their particular environment.

**Comment by krbz999** [comment]
Sure, that sounds good.

File: scripts/requestor.mjs [hunk 1]
```diff
@@ -74,5 +76,5 @@ export async function request({
     sound,
     flags: {[MODULE]: data, core: {canPopout: true}}
   };
```
Comment (krbz999) [inline scripts/requestor.mjs:78]:
```js
const messageData = foundry.utils.mergeObject(messageOptions, {
  content,
  whisper,
  sound,
  flags: {[MODULE]: data, core: {canPopout: true}}
});
return ChatMessage.create(messageData);
```

**Review by krbz999 — CHANGES_REQUESTED** [review]
Looks good to me. Just one small tweak.

**Pull request merged**